### ISSUE TITLE
[WARN] YCSB libraries not found. 

### ISSUE THREAD

**Issue by diogojfernandes** [brianfrankcooper/YCSB#1155]
Hello.

I get this message when trying to load the ycsb. It says "mvn is not recognized"  but as you can see on the print screen below I have maven installed and the environment variables created aswell. Also have "maven\bin" on PATH.

![image](https://user-images.githubusercontent.com/39376608/40494909-e6c35454-5f6d-11e8-93de-b7e63eb16eaa.png)

**Comment by diogojfernandes** [comment]
@busbey No, the 'basic' datastore also doesn't work.

I compiled the ArangoDB3 binding with success before loading the workload with 
`mvn -pl com.yahoo.ycsb:arangodb3-binding -am clean package`

I analysed the 'ycsb.bat' and it crashes on the line 172, but just call MAVEN on line 173.
What can I do?

**Comment by diogojfernandes** [comment]
@busbey when trying to execute
`./bin/ycsb.bat load basic -P workloads/workloada`

I get this error
![image](https://user-images.githubusercontent.com/39376608/40547630-5c6b2d64-602a-11e8-9002-7a483c421e9f.png)

**Comment by busbey** [comment]
does doing a full maven build from the top work? that error with the basic looks like you can't connect to maven central.

**Comment by diogojfernandes** [comment]
By "full maven build" you mean run this code?
`mvn clean package`

Should I delete the repository and reinstall it before doing that?

**Comment by busbey** [comment]
this step from the windows set up guide:

https://github.com/brianfrankcooper/YCSB/wiki/Prerequisites-for-Windows#first-build

```mvn package``` should suffice. you might need to tell Maven to check for things it thinks are missing via ```mvn -U package```

**Comment by diogojfernandes** [comment]
Running `mvn -U package` it successfully builds CORE YCSB and ArangoDB3 (the one I want to benchmark).

![image](https://user-images.githubusercontent.com/39376608/40551373-9a462620-6034-11e8-8733-ae3569714e2c.png)

 When I try to load **basic** or **arangodb3** datastore it fails building CORE YCSB

![image](https://user-images.githubusercontent.com/39376608/40551422-c55dfd06-6034-11e8-9559-d5a5aca601fd.png)

**Comment by busbey** [comment]
the build from the top level failing is a red flag that something is wrong in the build environment. what's the error in the first case?

**Comment by busbey** [comment]
looks like you should first create a database 'ycsb' with table 'usertable' in your ArangoDB instance.

**Comment by diogojfernandes** [comment]
I already did, same problem...
![image](https://user-images.githubusercontent.com/39376608/40578798-6ad37134-6112-11e8-8c5c-665be985a0ff.png)

On ArangoDb3 README it says to "drop the existing collection: "usertable" under database "ycsb"" with this code `db._collection("usertable").drop()` but it doesn't recognise "usertable".

On my comment above it says that failed to create table ycsb and collection usertable so my instance shouldn't have those created before (?).

Maybe I have to tell the default user login is **root**?

**Comment by busbey** [comment]
It's hard to say. The ArangoDB3 binding hasn't made it to our supported list yet, so it's not surprising that these kinds of edge cases could come up.

The error you got does look like a case of "I can't verify that you're authorized to do that", but I don't see anything in the current docs to say how to specify a user/password.

Does the older ArangoDB binding fail in the same way? It looks like it was tested in the 0.12.0 release.

**Comment by diogojfernandes** [comment]
Is it possible to benchmark ArangoDB3 with 0.12.0?

**Comment by busbey** [comment]
I don't think so, it was added in the 0.13.0 release. Does the ArangoDB binding work with different server versions? If so, we should document that.

**Comment by diogojfernandes** [comment]
I get the message "Unauthorized" when trying the ArangoDB binding. 
![image](https://user-images.githubusercontent.com/39376608/40606908-cb618228-625e-11e8-8a73-55da6fe66717.png)


I run the GitBash with admin privileges and set "Administrate" one ArangoDB root user for all databases.
![image](https://user-images.githubusercontent.com/39376608/40606878-af7394f2-625e-11e8-8ac6-0828628875bb.png)

**Comment by diogojfernandes** [comment]
@busbey , Once I get the error 401 Unauthorized, I searched for it and I found this:
https://httpstatuses.com/401

> The request has not been applied because it lacks valid authentication credentials for the target resource.

This means that maybe it neads the user login (**root**) on parameters?

**Comment by diogojfernandes** [comment]
SOLVED

I had to turn off authentication parameter on the **arangod.conf** file located on C:\ArangoDB\etc\arangodb3\arangod.conf 
`authentication = false`

**Comment by busbey** [comment]
Also, I'd like to have an open issue that tracks this gap so I can reference it in release notes. Would you prefer I reopen this one, or make a new one? (if I reopen this one you'll get github notifications when folks comment here or a PR fixes it, unless you unsubscribe from the issue.)

**Comment by diogojfernandes** [comment]
Not **add**, but **change**, it's value is "true" by default. Also, that information should be on ArangoDB3 README before the command `db._collection("usertable").drop()` One note: the file name is arangod.conf and not arangodb.conf.
 ```
The ArangoDB client bindings are not set up to provide authentication details to the ArangoDB instance, and as such you must ensure authentication is off for the system under test.

You can do this by altering your arangod.conf file and modify authentication parameter to "authentication=false".
```
You can reopen this ticket, no problem.
